xoai-common/src/main/java/io/gdcc/xoai/xml/CopyElement.java

```diff
@@ -14,7 +14,6 @@
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
-import java.util.regex.Matcher;
```

Comment:
I haven't run the code but yes, what @ErykKul is saying about thread safety seems well documented.

"Instances of this class are not safe for use by multiple concurrent threads." -- https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Matcher.html

"Instances of this class are immutable and are safe for use by multiple concurrent threads. Instances of the Matcher class are not safe for such use." -- https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html

So, yes, it sounds good to me to swap Matcher for Pattern!